improver/nowcasting/optical_flow.py

```diff
@@ -206,6 +209,93 @@ def __repr__(self):
             self.point_weight,
         )
 
+    @staticmethod
+    def _check_input_cubes(cube1, cube2):
+        """Check that input cubes have appropriate and matching dimensions"""
+        # check the nature of the input cubes, and raise a warning if they are
+        # not both precipitation
+        if cube1.name() != cube2.name():
+            msg = "Input cubes contain different data types {} and {}"
+            raise ValueError(msg.format(cube1.name(), cube2.name()))
+
+        data_name = cube1.name().lower()
+        if "rain" not in data_name and "precipitation" not in data_name:
+            msg = (
+                "Input data are of non-precipitation type {}.  Plugin "
+                "parameters have not been tested and may not be appropriate"
+                " for this variable."
+            )
+            warnings.warn(msg.format(cube1.name()))
+
+        # check cubes have exactly two spatial dimension coordinates and a
+        # scalar time coordinate
+        check_input_coords(cube1, require_time=True)
+        check_input_coords(cube2, require_time=True)
+
+        # check cube dimensions match
+        if cube1.coord(axis="x") != cube2.coord(axis="x") or cube1.coord(
+            axis="y"
+        ) != cube2.coord(axis="y"):
+            raise InvalidCubeError("Input cubes on unmatched grids")
+
+        # check grids are equal area
+        check_if_grid_is_equal_area(cube1)
+        check_if_grid_is_equal_area(cube2)
+
+    @staticmethod
+    def _get_advection_time(cube1, cube2):
+        """Get time over which the advection has occurred, in seconds, using the
+        difference in time or forecast reference time between input cubes"""
+        time_diff_seconds = (
+            cube2.coord("time").cell(0).point - cube1.coord("time").cell(0).point
+        ).total_seconds()
+        time_diff_seconds = int(time_diff_seconds)
+
+        if time_diff_seconds == 0:
+            # second cube should be an observation; first cube should have a
+            # non-zero forecast period which describes the advection time
+            if (
+                cube2.coords("forecast_period")
+                and cube2.coord("forecast_period").points[0] != 0
+            ):
+                raise InvalidCubeError(
+                    "Second input cube must be a current observation"
+                )
+
+            # get the time difference from the first cube's forecast period
+            fp_coord = cube1.coord("forecast_period").copy()
+            fp_coord.convert_units("seconds")
+            (time_diff_seconds,) = fp_coord.points
```

Comment:
What are the brackets here for out of interest?

Reply:
Black put them there!